mindmeld/components/entity_resolver.py

```diff
@@ -1544,40 +1121,136 @@ class TfIdfSparseCosSimEntityResolver(EntityResolverBase):
+        """
+        Analyzer that accounts for character ngrams as well as individual words in the input
+        """
+        # get char ngrams
+        results = self._char_ngrams_analyzer(string)
+        # add individual words
         words = re.split(r'[\s{}]+'.format(re.escape(punctuation)), string.strip())
         results.extend(words)
         return results
 
-    def _char_ngram_analyzer(self, string):
+    def _char_ngrams_analyzer(self, string):
+        """
+        Analyzer that only accounts for character ngrams from size 1 to self.ngram_length
+        """
+        string = string.strip()
+        if len(string) == 1:
+            return [string]
+
         results = []
-        # give more importance to starting and ending characters of a word
-        string = f" {string.strip()} "
+        # give importance to starting and ending characters of a word
+        string = f" {string} "
         for n in range(self.ngram_length + 1):
             results.extend([''.join(gram) for gram in zip(*[string[i:] for i in range(n)])])
         results = list(set(results))
         results.remove(' ')
         # adding lowercased single characters might add more noise
         results = [r for r in results if not (len(r) == 1 and r.islower())]
+        # returns empty list of an empty string
+        return results
+
+    def find_similarity(self,
+                        src_texts,
+                        top_n=20,
+                        scores_normalizer=None,
+                        _return_as_dict=False,
+                        _no_sort=False):
+        """Computes sparse cosine similarity
+
+        Args:
+            src_texts (Union[str, list]): string or list of strings to obtain matching scores for.
+            top_n (int, optional): maximum number of results to populate. if None, equals length
+                of self._syn_tfidf_matrix
+            scores_normalizer (str, optional): normalizer type to normalize scores. Allowed values
+                are: "min_max_scaler", "standard_scaler"
+           _return_as_dict (bool, optional): if the results should be returned as a dictionary of
+                target_text name as keys and scores as corresponding values
+            _no_sort (bool, optional): If True, results are returned without sorting. This is
+                helpful at times when you wish to do additional wrapper operations on top of raw
+                results and would like to save computational time without sorting.
+        Returns:
+            Union[dict, list[tuple]]: if _return_as_dict, returns a dictionary of tgt_texts and
+                their scores, else a list of sorted synonym names paired with their
+                similarity scores (descending order)
+        """
+
+        is_single = False
+        if isinstance(src_texts, str):
+            is_single = True
+            src_texts = [src_texts]
+
+        top_n = self._syn_tfidf_matrix.shape[0] if not top_n else top_n
```

Comment:
`top_n = len(self._syn_tfidf_matrix) if not top_n else top_n` should suffice, no?

Reply:
So sparse matrices don't have `len` attribute. So used shape instead.